src/rust/en.mangafox/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std]
+#![feature(pattern)]
+
+use aidoku::{
+	error::Result, prelude::*, std::net::HttpMethod, std::net::Request, std::String, std::Vec,
+	Chapter, Filter, Manga, MangaPageResult, Page,
+};
+
+mod parser;
+mod unpacker;
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let url = parser::get_filtered_url(filters, page);
+	println!("{}", &url);
+	let html = Request::new(url.as_str(), HttpMethod::Get)
+		.html()
+		.expect("");
+	parser::parse_directory(html)
+}
+
+#[get_manga_details]
+fn get_manga_details(manga_id: String) -> Result<Manga> {
+	let url = String::from("https://www.fanfox.net/manga/") + &manga_id;
+	let html = Request::new(url.as_str(), HttpMethod::Get)
+		.html()
+		.expect("");
+	parser::parse_manga(html, manga_id)
+}
+
+#[get_chapter_list]
+fn get_chapter_list(manga_id: String) -> Result<Vec<Chapter>> {
+	let url = String::from("https://www.fanfox.net/manga/") + &manga_id;
+	let html = Request::new(url.as_str(), HttpMethod::Get)
+		.header("Cookie", "isAdult=1")
+		.html()
+		.expect("");
+	parser::parse_chapters(html)
+}
+
+#[get_page_list]
+fn get_page_list(_manga_id: String, chapter_id: String) -> Result<Vec<Page>> {
+	let url = format!("https://m.fanfox.net/manga/{}", chapter_id);
```

Comment:
you should be consistent. probably use `format!` for the other two functions instead of string concating

Reply:
changed to format strings